tidy3d/components/data/sim_data.py

```diff
@@ -368,39 +560,37 @@ def plot_field(  # pylint:disable=too-many-arguments, too-many-locals, too-many-
             )
 
         # get the spatial coordinate corresponding to the plane
-        planar_coord = [name for name, val in spatial_coords_in_data.items() if val is False][0]
+        planar_coord = [name for name, c in spatial_coords_in_data.items() if c is False][0]
         axis = "xyz".index(planar_coord)
         position = float(field_data.coords[planar_coord])
 
-        # the frequency at which to evaluate the permittivity with None signaling freq -> inf
-        freq_eps_eval = sel_kwargs["f"] if "f" in sel_kwargs else None
-        return self.plot_field_array(
+        return self.plot_scalar_array(
             field_data=field_data,
             axis=axis,
             position=position,
-            val=val,
             freq=freq_eps_eval,
             eps_alpha=eps_alpha,
             robust=robust,
             vmin=vmin,
             vmax=vmax,
+            cmap_type=cmap_type,
             ax=ax,
         )
 
+    # pylint: disable=too-many-arguments,too-many-locals
     @equal_aspect
     @add_ax_if_none
-    # pylint:disable=too-many-arguments, too-many-locals, too-many-branches, too-many-statements
-    def plot_field_array(
+    def plot_scalar_array(
         self,
         field_data: xr.DataArray,
         axis: Axis,
         position: float,
-        val: Literal["real", "imag", "abs"] = "real",
         freq: float = None,
         eps_alpha: float = 0.2,
         robust: bool = True,
         vmin: float = None,
         vmax: float = None,
+        cmap_type: Literal["divergent", "sequential", "cyclic"] = "divergent",
```

Comment:
maybe better to define the cmap type in types.py along with FieldVal and PlotScale so they exist in the same place.